File: programmers/코딩테스트 고득점 Kit/완전탐색/모의고사.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> answers) {
	int a1 = 0, a2 = 0, a3 = 0, Max = 0;
	int arr1[10] = { 1,2,3,4,5, };
	int arr2[10] = { 2,1,2,3,2,4,2,5, };
	int arr3[10] = { 3,3,1,1,2,2,4,4,5,5 };
	for (int i = 0; i < answers.size(); i++) {
		if (answers[i] == arr1[i % 5]) a1++;
		if (answers[i] == arr2[i % 8]) a2++;
		if (answers[i] == arr3[i % 10]) a3++;
	}
	if (a1 > Max) Max = a1;
	if (a2 > Max) Max = a2;
	if (a3 > Max) Max = a3;
	vector<int> ret;
	if (a1 == Max) ret.push_back(1);
	if (a2 == Max) ret.push_back(2);
	if (a3 == Max) ret.push_back(3);
	return ret;
}

File: programmers/코딩테스트 고득점 Kit/완전탐색/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int Max = 0;
        int[] answer = {};
        int[] one = {1,2,3,4,5}; 
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three = {3,3,1,1,2,2,4,4,5,5};
        int[] score = new int[3];
        for(int i=0;i<answers.length;i++) {
            int aI = i % 5, bI = i % 8, cI = i % 10;
            if(one[aI] == answers[i]) score[0]++;
            if(two[bI] == answers[i]) score[1]++;
            if(three[cI] == answers[i]) score[2]++;
        }
        int cnt = 0;
        for(int i=0;i<3;i++)
            if(score[i] > Max) Max = score[i];
        for(int i=0;i<3;i++)
            if(score[i] == Max) cnt++;
        answer = new int[cnt];
        for(int i=0,j=0;i<3;i++)
            if(score[i] == Max) answer[j++] = i+1;
        return answer;
    }
}